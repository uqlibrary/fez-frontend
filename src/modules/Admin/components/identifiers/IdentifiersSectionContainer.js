import React from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';
import { connect } from 'react-redux';
import { getFormValues } from 'redux-form/immutable';

import { FORM_NAME } from '../../constants';
import { FormValuesContext } from 'context';
import IdentifiersSection from './IdentifiersSection';

export const IdentifiersSectionContainer = ({ disabled, formValues }) => {
    return (
        <FormValuesContext.Provider value={{ formValues: formValues.toJS() }}>
            <IdentifiersSection disabled={disabled} />
        </FormValuesContext.Provider>
    );
};

IdentifiersSectionContainer.propTypes = {
    disabled: PropTypes.bool,
    formValues: PropTypes.object,
};

export const mapStateToProps = (state, ownProps) => {
    const formValues = getFormValues(FORM_NAME)(state) || Immutable.Map({});
    return {
        disabled: ownProps.disabled,
        formValues: formValues.get('identifiersSection') || Immutable.Map({}),
    };
};

export default connect(mapStateToProps)(React.memo(IdentifiersSectionContainer));
