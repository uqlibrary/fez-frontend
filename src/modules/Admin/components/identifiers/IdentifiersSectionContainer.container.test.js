import { IdentifiersSectionContainer, mapStateToProps } from './IdentifiersSectionContainer';
import Immutable from 'immutable';

function setup(testProps = {}, args = { isShallow: true }) {
    const props = {
        formValues: {
            toJS: jest.fn(() => ({})),
        },
        ...testProps,
    };

    return renderComponent(IdentifiersSectionContainer, props, args);
}

describe('IdentifiersSectionContainer component', () => {
    it('should render default view', () => {
        const render = setup();
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should render disabled view', () => {
        const render = setup({ disabled: true });
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should map state to props', () => {
        expect(mapStateToProps({}, {})).toEqual({
            formValues: Immutable.Map({}),
        });
    });
});
