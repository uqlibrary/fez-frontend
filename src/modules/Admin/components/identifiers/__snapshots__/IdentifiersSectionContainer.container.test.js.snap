// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdentifiersSectionContainer component should render default view 1`] = `
<Context.Provider
  value={
    {
      "formValues": {},
    }
  }
>
  <Memo(IdentifiersSection) />
</Context.Provider>
`;

exports[`IdentifiersSectionContainer component should render disabled view 1`] = `
<Context.Provider
  value={
    {
      "formValues": {},
    }
  }
>
  <Memo(IdentifiersSection)
    disabled={true}
  />
</Context.Provider>
`;
